docs/reference/api.md

```diff
@@ -0,0 +1,214 @@
+# API
+
+Pebble exposes an HTTP API that remote clients can use to interact with the daemon. The API has endpoints for starting and stopping services, adding configuration layers to the plan, and so on.
+
+The API uses HTTP over a Unix socket, with access to the API controlled by user ID. If `pebble run` is started with the `--http <address>` option, Pebble exposes a limited set of open-access endpoints (see {ref}`api-access-levels`) using the given TCP address.
+
+## Structure of the API
+
+Here is a breakdown of the general structure of most API endpoints in Pebble.
+
+### API endpoint URL
+
+Pebble API uses standard HTTP URLs to get actions and resources. The URLs are made up of a base URL followed by a path and optional query parameters.
+
+- Base URL: The consistent foundation of all API endpoints (for example, `http://localhost:4000/v1/`).
```

Comment:
I think we can leave out the description "The consistent foundation of all API endpoints" and directly say what the base URL is likely to be in Pebble, especially because most endpoints use the socket. How about something like this:

```suggestion
- Base URL: If you're accessing the API over the Unix socket, the base URL is not applicable. If you're accessing the API over TCP, the base URL is `http://localhost:4000` by default (you can configure the TCP address using `pebble run --http`.
```

Reply:
Agreed on the text change. However, "the base URL" *path* part is applicable, just not the scheme and hostname. Perhaps "the scheme, hostname, and port are not applicable; only the `/v1/` path is important."